benchmark/multidimensional_memset_benchmark_kernels.cpp

```diff
@@ -0,0 +1,24 @@
+
+// Copyright (c) 2023 NVIDIA Corporation (reply-to: brycelelbach@gmail.com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <flux/op/cartesian_product.hpp>
+#include <flux/source/iota.hpp>
+#include <flux/op/for_each.hpp>
+
+void memset_2d_reference(double* A, std::size_t N, std::size_t M)
+{
+    for (std::size_t j = 0; j != M; ++j)
+        for (std::size_t i = 0; i != N; ++i)
+            A[i + j * N] = 0.0;
+}
+
+void memset_2d_flux_cartesian_product_iota(double* A, std::size_t N, std::size_t M)
+{
+    flux::cartesian_product(flux::iota(0LU, N), flux::iota(0LU, M))
```

Comment:
The `iota()` calls here are tripping up 32-bit Windows because `size_t` isn't unsigned long there, so we get ambiguous deduction (`iota()` expects the same type for both arguments)

```suggestion
    flux::cartesian_product(flux::iota(std::size_t{0}, N), flux::iota(std::size_t{0}, M))
```

Reply:
Changed this to `ints` and changed `N` and `M` to `flux::distance_t` in 4132087e2b68de4ad69c0643e1cbaff2b67b5ecd.